### ISSUE TITLE
devel-project: skip requests with pending factory-repo-checker review

### ISSUE THREAD

**Issue by jberry-suse** [openSUSE/osc-plugin-factory#802]
A reminder on a request like https://build.opensuse.org/request/show/482895 would seem suboptimal. If pending build failure in parent project which is noted by `factory-repo-checker` it is likely those requests should skip reminders.

Thoughts?

**Comment by jberry-suse** [comment]
Likely related, if a request has a staging review open like https://build.opensuse.org/request/show/486464.

**Comment by jberry-suse** [comment]
On the `factory-repo-checker` perhaps those requests should just be dropped after a period of days?

**Comment by lnussel** [comment]
we could limit the reminders to project[/package] ones for now. The main reason for forgotten reviews are the ones where the devel project is added as reviewer I think.

**Comment by lnussel** [comment]
this needs to be fixed before we can turn the reminder on again. Needs a dry run to double check :)

**Comment by nilxam** [comment]
I think it should ignore **factory-staging**, **factory-repo-checker** and **by_project to valid staging project** review

**Comment by jberry-suse** [comment]
Alright, after some digging I determined this was not caused for the reason I suspected, but rather the fact the `openSUSE:Factory` is included in the list of devel projects for `openSUSE:Factory`. I am not sure of the intent in setting the devel project in this manor. The result is that the `requests` sub-command ended up treating requests to `openSUSE:Factory` as requests to a devel project, which makes sense, but is not desirable. It then looked up the maintainers for the target project (had nothing to do with active reviews) and submitted a reminder for the result. Since no one maintains those packages in Factory it fell back to the project (ie Factory) who is maintained by the two referenced in the comment.

```
@dimstar_suse, @factory-repo-checker: review reminder
```

So in fact the code was behaving as design, but the input data was unexpected. I assumed the list of such packages was small, but I checked and it is not. As such I am going to assume this is in some way desired. For the script I can either leave `openSUSE:Factory` in the list placed in the dashboard and ignore when running the other two sub-commands or strip it when generating the list. The latter seems to make the most sense.

For reference:

- [list of such packages](https://gist.github.com/jberry-suse/21fb81d2a10fba2191e835365aa60c5a)
- [an example](https://build.opensuse.org/package/show/openSUSE:Factory/python-pytest-doc)

Re-evaluating the premise of this issue, it seems that ignoring such reviews was not the issue and likely does not make sense.

- If `factory-repo-checker` has not completed review someone should probably look at it.
- If staging review open likely irrelevant as still needs review and possible fixes.

The only one that may hold merit is `factory-staging` which indicates the request is not staged. If a request is passed the reminder deadline and not yet staged that likely means it is ignored or things are backed up. The ignored case should probably be singled out and checked directly if desired rather than depending on `factory-staging` review. If something is ignored it is likely waiting on something else, but at the point reason for ignored is resolved the review will be desired so this likely is not something the reminder script should care about either.

TL;DR Removing `openSUSE:Factory` from devel list (which acts like corrupt data) should make the script behave as intended. It does not seem necessary to make additional changes.

**Comment by lnussel** [comment]
the project self reference happens on package links for packages with second spec file

**Comment by jberry-suse** [comment]
Ah, that makes sense. Then stripping it out definitely makes sense.

**Comment by jberry-suse** [comment]
Running `requests` today the list is much smaller as expected. Not sure if we want to enable again as that seems to be the only issue. I've spot checked a number of requests and they seem fine.
